ps_democqrshooksusage.php

```diff
@@ -0,0 +1,337 @@
+                (new ToggleColumn('is_allowed_for_review'))
+                    ->setName($translator->trans('Allowed for review', [], 'Modules.Ps_DemoCQRSHooksUsage'))
+                    ->setOptions([
+                        'field' => 'is_allowed_for_review',
+                        'primary_field' => 'id_customer',
+                        'route' => 'ps_democqrshooksusage_toggle_is_allowed_for_review',
+                        'route_param_name' => 'customerId',
+                    ])
+            )
+        ;
+
+        $definition->getFilters()->add(
+            (new Filter('is_allowed_for_review', YesAndNoChoiceType::class))
+            ->setAssociatedColumn('is_allowed_for_review')
+        );
+    }
+
+    /**
+     * Hook allows to modify Customers query builder and add custom sql statements.
+     *
+     * @param array $params
+     */
+    public function hookActionCustomerGridQueryBuilderModifier(array $params)
+    {
+        /** @var QueryBuilder $searchQueryBuilder */
+        $searchQueryBuilder = $params['search_query_builder'];
+
+        /** @var CustomerFilters $searchCriteria */
+        $searchCriteria = $params['search_criteria'];
+
+        $searchQueryBuilder->addSelect(
+            'IF(dcur.`is_allowed_for_review` IS NULL,0,dcur.`is_allowed_for_review`) AS `is_allowed_for_review`'
+        );
+
+        $searchQueryBuilder->leftJoin(
+            'c',
+            '`' . pSQL(_DB_PREFIX_) . 'democqrshooksusage_reviewer`',
+            'dcur',
+            'dcur.`id_customer` = c.`id_customer`'
+        );
+
+        if ('is_allowed_for_review' === $searchCriteria->getOrderBy()) {
+            $searchQueryBuilder->orderBy('dcur.`is_allowed_for_review`', $searchCriteria->getOrderWay());
+        }
+
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if ('is_allowed_for_review' === $filterName) {
+                $searchQueryBuilder->andWhere('dcur.`is_allowed_for_review` = :is_allowed_for_review');
+                $searchQueryBuilder->setParameter('is_allowed_for_review', $filterValue);
+
+                if (!$filterValue) {
+                    $searchQueryBuilder->orWhere('dcur.`is_allowed_for_review` IS NULL');
+                }
+            }
+        }
+    }
+
+    /**
+     * Hook allows to modify Customers form and add aditional form fields as well as modify or add new data to the forms.
+     *
+     * @param array $params
+     */
+    public function hookactioncustomerFormBuilderModifier(array $params)
+    {
+        /** @var FormBuilderInterface $formBuilder */
+        $formBuilder = $params['form_builder'];
+        $formBuilder->add('is_allowed_for_review', SwitchType::class);
+
+        /** @var CommandBusInterface $queryBus */
+        $queryBus = $this->get('prestashop.core.query_bus');
+
+        /** @var ReviewerSettingsForForm $reviewerSettings */
+        $reviewerSettings = $queryBus->handle(new GetReviewerSettingsForForm($params['id']));
```

Comment:
Let's remind here that we choose to use a Command in order to demonstrate how it can be used, but this is not mandatory.